Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp

```diff
@@ -126,8 +127,14 @@ void Wiimote::HidOutputReport(const wm_report* const sr, const bool send_ack)
     break;
 
   case RT_WRITE_SPEAKER_DATA:  // 0x18
+    // Not sure if speaker mute stops the bus write on real hardware, but it's not that important
     if (!m_speaker_mute)
-      Wiimote::SpeakerData(reinterpret_cast<const wm_speaker_data*>(sr->data));
+    {
+      auto sd = reinterpret_cast<const wm_speaker_data*>(sr->data);
+      if (sd->length > 20)
+        PanicAlert("EmuWiimote: bad speaker data length!");
+      m_i2c_bus.BusWrite(0x51, 0x00, sd->length, sd->data);
```

Comment:
Magic Number?

Reply:
It's the speaker peripheral slave address. I should use some constants.